app/src/main/java/space/taran/arknavigator/mvp/model/repo/tags/PlainTagsStorage.kt

```diff
@@ -230,21 +236,19 @@ class PlainTagsStorage(
             val lines = Files.readAllLines(storageFile, StandardCharsets.UTF_8)
             verifyVersion(lines.removeAt(0))
 
-            val result = lines
-                .map {
+            val result = mutableMapOf<ResourceId, Tags>()
+
+            try {
+                lines.map {
                     val parts = it.split(KEY_VALUE_SEPARATOR)
                     val id = ResourceId.fromString(parts[0])
                     val tags = tagsFromString(parts[1])
-
-                    if (tags.isEmpty()) {
-                        throw AssertionError(
-                            "Tags storage must not contain empty sets of tags"
-                        )
-                    }
-
-                    id to tags
+                    isCorrupted = tags.isEmpty()
```

Comment:
If `tags.isEmpty()` we can also `break` the loop in addition to setting `isCorrupted` flag.